pandas/core/indexes/base.py

```diff
@@ -1376,6 +1383,12 @@ def set_names(self, names, level=None, inplace: bool = False):
         if not is_list_like(names) and level is None and self.nlevels > 1:
             raise TypeError("Must pass list-like as `names`.")
 
+        if is_dict_like(names) and not isinstance(self, ABCMultiIndex):
+            raise TypeError("Can only pass dict-like as `names` for MultiIndex.")
```

Comment:
testing these conditions?

Reply:
Yep, in ``test_name_with_dict_like_raising``